src/core/processing/models/qgsprocessingmodelresult.h

```diff
@@ -0,0 +1,157 @@
+
+#ifndef QGSPROCESSINGMODELRESULT_H
+#define QGSPROCESSINGMODELRESULT_H
+
+#include "qgis_core.h"
+#include "qgis.h"
+
+/**
+ * \ingroup core
+ * \brief Encapsulates the results of running a child algorithm within a model
+ *
+ * \since QGIS 3.38
+*/
+class CORE_EXPORT QgsProcessingModelChildAlgorithmResult
+{
+  public:
+
+    QgsProcessingModelChildAlgorithmResult();
+
+    /**
+     * Returns the status of executing the child algorithm.
+     *
+     * \see setExecutionStatus()
+     */
+    Qgis::ProcessingModelChildAlgorithmExecutionStatus executionStatus() const { return mExecutionStatus; }
+
+    /**
+     * Sets the \a status of executing the child algorithm.
+     *
+     * \see executionStatus()
+     */
+    void setExecutionStatus( Qgis::ProcessingModelChildAlgorithmExecutionStatus status ) { mExecutionStatus = status; }
+
+    /**
+     * Returns the inputs used for the child algorithm.
+     *
+     * \see setInputs()
+     */
+    QVariantMap inputs() const { return mInputs; }
+
+    /**
+     * Sets the \a inputs used for the child algorithm.
+     *
+     * \see inputs()
+     */
+    void setInputs( const QVariantMap &inputs ) { mInputs = inputs; }
+
+    /**
+    * Returns the outputs generated by the child algorithm.
+    *
+    * \see setOutputs()
+    */
+    QVariantMap outputs() const { return mOutputs; }
+
+    /**
+     * Sets the \a outputs used for the child algorithm.
+     *
+     * \see outputs()
+     */
+    void setOutputs( const QVariantMap &outputs ) { mOutputs = outputs; }
+
+    /**
+     * Returns the HTML formatted contents of logged messages which occurred while running the child.
+     *
+     * \see setHtmlLog()
+     */
+    QString htmlLog() const { return mHtmlLog; }
+
+    /**
+     * Sets the HTML formatted contents of logged messages which occurred while running the child.
+     *
+     * \see htmlLog()
+     */
+    void setHtmlLog( const QString &log ) { mHtmlLog = log; }
+
+    bool operator==( const QgsProcessingModelChildAlgorithmResult &other ) const
+    {
+      return mExecutionStatus == other.mExecutionStatus
+             && mHtmlLog == other.mHtmlLog
+             && mInputs == other.mInputs
+             && mOutputs == other.mOutputs;
+    }
+    bool operator!=( const QgsProcessingModelChildAlgorithmResult &other ) const
+    {
+      return !( *this == other );
+    }
+
+  private:
+
+    Qgis::ProcessingModelChildAlgorithmExecutionStatus mExecutionStatus = Qgis::ProcessingModelChildAlgorithmExecutionStatus::NotExecuted;
+    QVariantMap mInputs;
+    QVariantMap mOutputs;
+    QString mHtmlLog;
+
+};
+
+/**
+ * \ingroup core
+ * \brief Encapsulates the results of running a Processing model
+ *
+ * \since QGIS 3.38
+*/
+class CORE_EXPORT QgsProcessingModelResult
+{
+  public:
+
+    QgsProcessingModelResult();
+
+
+    /**
+     * Returns the map of child algorithm results.
+     *
+     * Map keys refer to the child algorithm IDs.
+     *
+     * \since QGIS 3.38
+     */
```

Comment:
Since it's a new class, you don't need the \since.

```suggestion
    /**
     * Returns the map of child algorithm results.
     *
     * Map keys refer to the child algorithm IDs.
     */
```